craft_application/application.py

```diff
@@ -366,6 +369,26 @@ def _get_dispatcher(self) -> craft_cli.Dispatcher:
 
         return dispatcher
 
+    def _get_app_plugins(self) -> dict[str, PluginType]:
+        """Get the plugins for this application.
+
+        Should be overridden by applications that need to register plugins at startup.
+        """
+        return {}
+
+    def register_plugins(self, plugins: dict[str, PluginType]) -> None:
```

Comment:
Registering plugins feels more like the responsibility of the LifecycleService than of the Application itself. Do we even need to register the plugins if we're not starting up lifecycle commands? (I'm thinking no, but I'm unsure when it comes to validating the parts section of an `<appname>.yaml` file.)

Reply:
That's what @tigarmo asked, since rockcraft has something to do with plugins befroe the lifecycle begins. I think it was linting?

Reply:
We definitely validate the `parts` section when loading the project; [I am almost certain](https://github.com/canonical/craft-parts/blob/d214d9d8ef9606cfd3f46a139c220b43b082cf41/craft_parts/parts.py#L656) (but haven't tried) that a plugin that is used but isn't registered yet would throw an error. This wouldn't be a problem today for Rockcraft specifically, but it would for Snapcraft eventually (and maybe Charmcraft)?

Reply:
What I mean is that I'm fine with moving this somewhere else, but this "somewhere else" still needs to happen "early enough"

Reply:
Well, I moved the logic to `run()` that should be early enough